src/limb.rs

```diff
@@ -110,31 +146,70 @@ impl <'a> Range<'a> {
                 let mut dest_as_bytes = limbs_as_bytes_mut(dest);
                 try!(rng.fill(&mut dest_as_bytes));
             }
-            dest[self.max_exclusive.len() - 1] &= most_significant_limb_mask;
+            dest[self.max_exclusive.len() - 1] &= self.sampling_most_sig_limb_mask;
 
             if self.are_limbs_within(&dest) {
                 return Ok(());
             }
+
+            if self.sampling_reduce_when_over_bound {
```

Comment:
It may be worth putting this in its own function so that (a) we can test it, and (b) to reduce the indention for the giant comment.